File: Tutoriels/Java/POO/z0_Ressources/src/manager/GProcess.java
//===============================================
package manager;
//===============================================
public abstract class GProcess {
    //===============================================
    public static GProcess Instance() {
        String lProcess = GConfig.Instance().getData("PROCESS");
        if(lProcess == "HELLO") return GProcessHello.Instance();
        return GProcessHello.Instance();
    }
    //===============================================
    public abstract void run(String[] args);
    //===============================================
}
//===============================================

File: Tutoriels/Cpp/POO/z0_Ressources/src/manager/GProcess.cpp
//===============================================
#include "GProcess.h"
#include "GProcessHello.h"
#include "GConfig.h"
//===============================================
GProcess::GProcess() {

}
//===============================================
GProcess::~GProcess() {

}
//===============================================
GProcess* GProcess::Instance() {
    string lProcess = GConfig::Instance()->getData("PROCESS");
    if(lProcess == "HELLO") return GProcessHello::Instance();
    return GProcessHello::Instance();
}
//===============================================
void GProcess::run(int argc, char **argv) {}
//===============================================
